components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1827,6 +1829,32 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         return subscription;
     }
 
+    async getStripePublishableKey(ctx: TraceContext): Promise<string | undefined> {
```

Comment:
This method should also be behind a feature flag. The feature flag on the client is useless if we do not check it on the server side too.

Reply:
Thanks, but in this case I would disagree:
- Gating this behind the Usage-Based Feature flag would be over-engineered here (requires making yet another call to configcat from the server, and wait for the response which can take a while)
- We primarily want to hide Usage-Based Billing from users because it's work-in-progress, but if someone is dedicated enough to call the server API directly, I guess they're allowed to try out our Stripe integration (which we're planning to GA in a few weeks anyway)

Reply:
> We primarily want to hide Usage-Based Billing from users because it's work-in-progress, but if someone is dedicated enough to call the server API directly, I guess they're allowed to try out our Stripe integration (which we're planning to GA in a few weeks anyway)

The client must not be authoritative. We are not just hiding UI features, we're gatting functinality to control how we roll it out. We don't want anyone without the correct permissions (or membership) to access this feature - this cannot be done on the UI and must be done on the server side.

> making yet another call to configcat from the server, and wait for the response which can take a while)

Feature flags are updated async, actual eval is non-blocking.

Reply:
@jankeromnes I agree with @easyCZ : Deciding whether or not to enable a feature is a authorative (server-side) question. IMO it's the only way to serve this as "clean cut".
And whether or not we do another request against configcat should not guide any decision I think. :+1:

Reply:
Done ✅

Reply:
Should also be added for the other API methods that are for usage based billing.